### DISCUSSION TITLE
Welcome to pychoco Discussions!

### DISCUSSION THREAD

**Discussion by cprudhom** [chocoteam/pychoco#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dimitri-justeau** [comment]
It's been almost a year since we released pychoco !

According to PePy (https://pepy.tech/project/pychoco), pychoco has been downloaded more than 8500 times since its release. However, I did not seen any feedback about this beta release. Maybe we have not communicated enough about it ? Maybe it is not useful ? I do not know.

If we want to continue with this project, the good news is that Oracle GraalVM (previously known as Oracle GraalVM enterprise) is now FREE !

https://medium.com/graalvm/a-new-graalvm-release-and-new-free-license-4aab483692f5

One of the main differences with the community edition that we used for pychoco 0.1 is that the native-image component is much more efficient, both in terms of memory and computing performances. It also includes the profile-guided compiler optimizations. I think that we should try to leverage this, and evaluate the performances of a native-image of Choco with this new version.

**Comment by cprudhom** [comment]
I think we can define a roadmap for immediate actions (to be carried out, say, at the beginning of July ;) ) and longer-term actions.

**Comment by dimitri-justeau** [comment]
Yes I'm in :-) I'll prepare something before the end of the week !

**Comment by dimitri-justeau** [comment]
Some ideas for immediate actions:

- [x] Update pychoco (and choco-solver-capi) with last Choco's version. 
- [x] Compile choco-solver-capi with last version of GraalVM.
- Fix issues currently opened in Github.
    - [x] Display variables' domain on print.
    - [x] Allow enumerated IntVars.
    - [ ]  Implement IReificationFactory (only ifthen ifthenelse) ?
- [ ] Design notebook examples of pychoco.
- [ ] Check research strategies
- [ ] Check parallel portfolio
- [ ] Bench some problems to assess performances
